File: api/src/main/java/com/ruby/rubia_server/core/dto/CreateCompanyDTO.java
package com.ruby.rubia_server.core.dto;

import com.ruby.rubia_server.core.enums.CompanyPlanType;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CreateCompanyDTO {

    @NotBlank(message = "Nome é obrigatório")
    @Size(min = 2, max = 255, message = "Nome deve ter entre 2 e 255 caracteres")
    private String name;

    @NotBlank(message = "Slug é obrigatório")
    @Pattern(regexp = "^[a-z0-9-]+$", message = "Slug deve conter apenas letras minúsculas, números e hífens")
    @Size(min = 2, max = 255, message = "Slug deve ter entre 2 e 255 caracteres")
    private String slug;

    @Size(max = 1000, message = "Descrição não pode exceder 1000 caracteres")
    private String description;

    @Email(message = "Email de contato deve ter formato válido")
    private String contactEmail;

    @Pattern(regexp = "^\\+?[0-9.()\\-\\s]*$", message = "Telefone de contato inválido")
    private String contactPhone;

    private String logoUrl;

    @Builder.Default
    private Boolean isActive = true;

    @Builder.Default
    private CompanyPlanType planType = CompanyPlanType.BASIC;

    @Builder.Default
    private Integer maxUsers = 10;

    @Builder.Default
    private Integer maxWhatsappNumbers = 1;

    @NotNull(message = "ID do grupo da empresa é obrigatório")
    private UUID companyGroupId;
}

File: api-dotnet/Rubia.Server/DTOs/CreateCompanyDto.cs
using System.ComponentModel.DataAnnotations;
using System.Text.RegularExpressions;
using Rubia.Server.Enums;

namespace Rubia.Server.DTOs;

public class CreateCompanyDto : IValidatableObject
{
    [Required(ErrorMessage = "Nome é obrigatório")]
    [StringLength(255, MinimumLength = 2, ErrorMessage = "Nome deve ter entre 2 e 255 caracteres")]
    public string Name { get; set; } = string.Empty;

    [Required(ErrorMessage = "Slug é obrigatório")]
    [StringLength(255, MinimumLength = 2, ErrorMessage = "Slug deve ter entre 2 e 255 caracteres")]
    public string Slug { get; set; } = string.Empty;

    [StringLength(1000, ErrorMessage = "Descrição não pode exceder 1000 caracteres")]
    public string? Description { get; set; }

    [EmailAddress(ErrorMessage = "Email de contato deve ter formato válido")]
    public string? ContactEmail { get; set; }

    public string? ContactPhone { get; set; }

    public string? LogoUrl { get; set; }

    public bool IsActive { get; set; } = true;

    public CompanyPlanType PlanType { get; set; } = CompanyPlanType.Basic;

    public int MaxUsers { get; set; } = 10;

    public int MaxWhatsappNumbers { get; set; } = 1;

    public int MaxAiAgents { get; set; } = 1;

    [Required(ErrorMessage = "ID do grupo da empresa é obrigatório")]
    public Guid CompanyGroupId { get; set; }

    public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
    {
        var results = new List<ValidationResult>();

        // Validar slug
        if (!string.IsNullOrEmpty(Slug))
        {
            var slugRegex = new Regex(@"^[a-z0-9-]+$");
            if (!slugRegex.IsMatch(Slug))
            {
                results.Add(new ValidationResult(
                    "Slug deve conter apenas letras minúsculas, números e hífens",
                    new[] { nameof(Slug) }));
            }
        }

        // Validar telefone
        if (!string.IsNullOrEmpty(ContactPhone))
        {
            var phoneRegex = new Regex(@"^\+?[0-9.()-\s]*$");
            if (!phoneRegex.IsMatch(ContactPhone))
            {
                results.Add(new ValidationResult(
                    "Telefone de contato inválido",
                    new[] { nameof(ContactPhone) }));
            }
        }

        return results;
    }
}